pkg/pod/script.go

```diff
@@ -116,117 +105,125 @@ func convertScripts(shellImageLinux string, shellImageWin string, steps []v1beta
 	convertedStepContainers := convertListOfSteps(steps, &placeScriptsInit, &placeScripts, breakpoints, "script")
 
 	// Pass empty breakpoint list in "sidecar step to container" converter to not rewrite the scripts and add breakpoints to sidecar
-	sidecarContainers := convertListOfSteps(sideCarSteps, &placeScriptsInit, &placeScripts, []string{}, "sidecar-script")
+	sidecarContainers := convertListOfSidecars(sidecars, &placeScriptsInit, &placeScripts, "sidecar-script")
 	if placeScripts {
 		return &placeScriptsInit, convertedStepContainers, sidecarContainers
 	}
 	return nil, convertedStepContainers, sidecarContainers
 }
 
-// convertListOfSteps does the heavy lifting for convertScripts.
-//
-// It iterates through the list of steps (or sidecars), generates the script file name and heredoc termination string,
-// adds an entry to the init container args, sets up the step container to run the script, and sets the volume mounts.
-func convertListOfSteps(steps []v1beta1.Step, initContainer *corev1.Container, placeScripts *bool, breakpoints []string, namePrefix string) []corev1.Container {
-	containers := []corev1.Container{}
-	for i, s := range steps {
-		// Add debug mounts if breakpoints are present
-		if len(breakpoints) > 0 {
-			debugInfoVolumeMount := corev1.VolumeMount{
-				Name:      debugInfoVolumeName,
-				MountPath: filepath.Join(debugInfoDir, fmt.Sprintf("%d", i)),
-			}
-			steps[i].VolumeMounts = append(steps[i].VolumeMounts, debugScriptsVolumeMount, debugInfoVolumeMount)
-		}
-
-		if s.Script == "" {
-			// Nothing to convert.
-			containers = append(containers, *steps[i].ToK8sContainer())
-			continue
-		}
-
-		// Check for a shebang, and add a default if it's not set.
-		// The shebang must be the first non-empty line.
-		cleaned := strings.TrimSpace(s.Script)
-		hasShebang := strings.HasPrefix(cleaned, "#!")
-		requiresWindows := strings.HasPrefix(cleaned, "#!win")
-
-		script := s.Script
-		if !hasShebang {
-			script = defaultScriptPreamble + s.Script
-		}
+func placeScriptInContainer(script, scriptFile string, c *corev1.Container, initContainer *corev1.Container, placeScripts *bool) {
+	if script == "" {
+		return
+	}
+	cleaned := strings.TrimSpace(script)
+	hasShebang := strings.HasPrefix(cleaned, "#!")
+	requiresWindows := strings.HasPrefix(cleaned, "#!win")
 
-		// At least one step uses a script, so we should return a
-		// non-nil init container.
-		*placeScripts = true
+	if !hasShebang {
+		script = defaultScriptPreamble + script
+	}
+	*placeScripts = true
```

Comment:
instead of passing this pointer to a bool around, it might be more readable to do an initial loop over the containers to determine if any of them have scripts and therefore a scripts-init container is needed